src/libDirectoryService/DSBlockPreProcessing.cpp

```diff
@@ -509,6 +503,22 @@ bool DirectoryService::VerifyPoWOrdering(
   std::vector<PubKey> removeDSNodePubkeys =
       m_pendingDSBlock->GetHeader().GetDSRemovePubKeys();
 
+  // Sort and trim the PoW solutions
+  auto sortedPoWSolns =
+      SortPoWSoln(priorityNodePoWs, true, removeDSNodePubkeys.size());
+
+  // Remove the DS solutions from the PoW solutions.
```

Comment:
Need to check with you on this portion, if it's needed. I just made sure the same computation was done for the Backup nodes like in the Leader.